docs/incremental.md

```diff
@@ -0,0 +1,133 @@
+# Incremental Processing
+
+Incremental processing is a way to avoid re-processing of sources as much as possible.
+The major goal is to reduce the turn-around time of a typical change-compile-test cycle.
+Here is a [wiki link](https://en.wikipedia.org/wiki/Incremental_computing) to the more general
+idea of incremental computation.
+
+To be able to know which sources are dirty, i.e., those that need to be reprocessed, KSP needs
+processors' help to identify the correspondence of input sources and generated outputs.
+Because it is cumbersome and error prone to keep track of which inputs are involved in generating
+which outputs, KSP is designed to help with that and only require a minimum set of
+**sources that serve as roots that processors start to navigate the code structure**. In other
+words, a processor needs to associate an output with sources of those `KSNode`, if obtained from:
+* `Resolver.getAllFiles`
+* `Resolver.getSymbolsWithAnnotation`
+* `Resolver.getClassDesclarationByName`
+
+Currently, only changes in Kotlin and Java sources are tracked. If there is a change in the
+classpath, namely in other modules or libraries, a full re-processing will be triggered.
+
+Incremental processing is currently disabled by default. To enable it, set the Gradle property
+`ksp.incremental=true`. To enable logs, which dump the dirty set according to dependencies and
+outputs, use `ksp.incremental.log=true`. They can be found as `build/*.log`.
+
+## Aggregating v.s. Isolating
+The idea is similar but slightly different to the [definition](https://docs.gradle.org/current/userguide/java_plugin.html#sec:incremental_annotation_processing)
+in Gradle annotation processing. In KSP,
+* *aggregating* / *isolating* is associated with each output, rather than the entire processor
+* an isolating output can have several sources
+* aggregating means that an output can be affected by any changes
+
+If an output is `aggregating`, any changes except removal of files may affect it potentially.
```

Comment:
I think removal of file can also cause compile information change and should be reprocessed. Example: function type refined by a more concrete signature, and removal of that will change its type at call site.

Reply:
done